Assets/_Laz/Scripts/Environment/Puzzles/LazoActivation.cs

```diff
@@ -0,0 +1,58 @@
+using System.Linq;
+
+namespace Laz
+{
+    [System.Serializable]
+    public struct LazoActivation
+    {
+        public BasePuzzleBehaviour[] PuzzleItems;
+        public BaseCompletedPuzzleActivationBehaviour ItemToActivate;
+
+        public void Initialize()
+        {
+            foreach (var puzzleItem in PuzzleItems)
+            {
+                puzzleItem.Initialize();
+                puzzleItem.OnPuzzleCompleted += ActivateWrapItemIfNeeded;
+            }
+            
+            ItemToActivate.Initialize();
+        }
+
+        public void CleanUp()
+        {
+            foreach (var item in PuzzleItems)
+            {
+                item.CleanUp();
```

Comment:
should you also have item.OnPuzzleCompleted -= ActivateWrapItemIfNeeded

Reply:
Because this is a pure C# class, Garbage collector should remove it automatically when nulling it out.

Reply:
However, i'm looking into this more. to double check i might be incorrect and need to handle subscribing

Reply:
I set it up to unsubscribe.